tests/unit/test_routes/test_resource_create.py

```diff
@@ -0,0 +1,249 @@
+                           headers={'x-apikey': apikey})
+    assert_wrong_type(response, "object")
+
+    response = client.post('/api/v1/resources',
+                           json=1,
+                           headers={'x-apikey': apikey})
+    assert_wrong_type(response, "int")
+
+    response = client.post('/api/v1/resources',
+                           json=1.2,
+                           headers={'x-apikey': apikey})
+    assert_wrong_type(response, "number")
+
+    response = client.post('/api/v1/resources',
+                           json=True,
+                           headers={'x-apikey': apikey})
+    assert_wrong_type(response, "boolean")
+
+    response = client.post('/api/v1/resources',
+                           json="This should be a list",
+                           headers={'x-apikey': apikey})
+    assert_wrong_type(response, "string")
+
+
+def test_false_validation(module_client,
+                          module_db,
+                          fake_auth_from_oc,
+                          fake_algolia_save,
+                          fake_validation):
+    # Given the validate_resource method fails to catch errors
+    # When we commit the resource to the database
+    # Then the api returns a 422 response
+
+    client = module_client
+    first_term = random_string()
+    apikey = get_api_key(client)
+
+    resource = client.post("/api/v1/resources",
+                           json=[dict(
+                               name=f"{first_term}",
+                               category="Website",
+                               url=f"{first_term}",
+                               paid=False,
+                           )],
+                           headers={'x-apikey': apikey}
+                           )
+
+    assert (resource.status_code == 200)
+
+    # Type Error, json is dict when expecting list
+    response = client.post("/api/v1/resources",
+                           json=dict(
+                               name=f"{first_term}",
+                               category="Website",
+                               url=f"{first_term}",
+                               paid=False,
+                           ),
+                           headers={'x-apikey': apikey}
+                           )
+
+    assert_invalid_body(response)
+
+    # Attempting to update a resource with a non-unique URL.
+    response = client.put(f"/api/v1/resources/{1}",
+                          json=dict(
+                              name="New name",
+                              languages=["New language"],
+                              category="New Category",
+                              url=f"{first_term}",
+                              paid=False,
+                              notes="New notes"
+                          ),
+                          headers={'x-apikey': apikey}
+                          )
+    assert_correct_validation_error(response, ["url"])
```

Comment:
@aaron-suarez  This test is expecting for the validation to pass and the commit to the db to fail. When we fail on the db step, the response should be a generic 422 response, not the validation message built by the validation code. So if a "url" validation error exists, we arent testing the scenario that was described in the comment at the top of the test.

This is what I was talking about when I said I couldnt get the mocker to work. I should be mocking the validate resource calls but it looks like it is triggering anyways.

Reply:
Yeah, I spent about 2 hours working on this and I can't understand why it will mock the validate_resource call on the `POST` request but not on the `PUT` request...

I'm not really sure what to do to be honest, but I'm tired of messing with it to be honest.

Reply:
Oh I'm glad I wasn't doing something wrong at least.  I'm okay with this for now. It isn't a very likely scenario anyway.